pkg/db/jsonType.go

```diff
@@ -0,0 +1,60 @@
+package db
+
+import (
+	"database/sql"
+	"database/sql/driver"
+	"encoding/json"
+	"errors"
+
+	"github.com/justtrackio/gosoline/pkg/mdl"
+)
+
+var (
+	_                        driver.Valuer = JSONType[struct{}]{}
+	_                        sql.Scanner   = &JSONType[struct{}]{}
+	ErrJSONInvalidDriverType               = errors.New("invalid incoming driver type for json type")
+)
+
+type (
+	JSONType[T any] struct {
```

Comment:
I am unsure about the name of the type - it is a JSON value after all, not a type. And as a value, it should behave like any other value, right? So if I want to use it in a nullable column, I use a pointer to it, not have a flag on the actual value (which btw doesn't survive getting stored in the database - `Scan` never sets it)